modules/core/src/main/scala/org/tessellation/infrastructure/trust/TrustStorageUpdater.scala

```diff
@@ -0,0 +1,39 @@
+package org.tessellation.infrastructure.trust
+
+import cats.effect.std.Supervisor
+import cats.effect.{Async, Ref}
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import scala.concurrent.duration.DurationInt
+
+import org.tessellation.domain.trust.storage.TrustStorage
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.gossip.Gossip
+
+trait TrustStorageUpdater[F[_]] {
+  def update: F[Unit]
+}
+
+object TrustStorageUpdater {
+
+  def make[F[_]: Async](
+    getOrdinal: Ref[F, Option[SnapshotOrdinal]],
+    gossip: Gossip[F],
+    storage: TrustStorage[F]
+  ): TrustStorageUpdater[F] = new TrustStorageUpdater[F] {
+    override def update: F[Unit] = for {
+      maybeOrdinal <- getOrdinal.get
+      _ <- maybeOrdinal.traverse_(storage.updateCurrent) // This must come first.
```

Comment:
why are you using `traverse_`?

Reply:
This is at the recommendation of @sdavidp . The `traverse_` is used for only side effects. @sdavidp  can explain the reasoning better.

Reply:
The result is discarded so `traverse_` seems more appropriate than `traverse`. This is what is in the method comment:
_This method is primarily useful when G[_] represents an action or effect, and the specific A aspect of G[A] is not otherwise needed._

Reply:
:+1: